src/osd/ECCache.hpp

```diff
@@ -0,0 +1,273 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2016 Western Digital Corporation.
+ * Author: Tomy Cheru <tomy.cheru@sandisk.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#ifndef ECCACHE_HPP_INCLUDED
+#define ECCACHE_HPP_INCLUDED
+
+#include <iostream>
+//#include <string>
+#include <set>
+#include "include/buffer.h"
+#include "common/hobject.h"
+#include "common/Mutex.h"
+#include "common/Cond.h"
+
+class ecc_extent
+{
+  private:
+    unsigned eoff;
+    bufferlist edata;
+
+  public:
+    bool operator < (const ecc_extent& r) const
+    {
+      return eoff < r.eoff;
+    }
+
+    bool operator == (const ecc_extent& r) const
+    {
+      return eoff == r.eoff;
+    }
+
+    ecc_extent(unsigned o = 0,bufferlist d = {} )
```

Comment:
Also, don't use default arguments here.  For example: ecc_extent(10) makes no sense and is invalid anyway.  In the declaration of eoff, do
```
  unsigned eoff = 0;
```

Write the default and copy constructors as:
```
  ecc_extent() = default;
  ecc_extent(const ecc_extent &) = default
```

and add this constructor as
```
  ecc_extent(unsigned o, bufferlist d) : o(o), d(d) {
    // If you want, you can add:
    assert(o == d.length());
  }
```